### ISSUE TITLE
How Can I Build For Android

### ISSUE THREAD

**Issue by Igamegum** [Maratyszcza/NNPACK#65]
Under Linux

**Comment by Maratyszcza** [comment]
https://github.com/Maratyszcza/NNPACK#cross-compilation-for-android

**Comment by Igamegum** [comment]
Navigate to NNPACK directory and setup dependencies (confu setup)
What does this mean?

**Comment by Igamegum** [comment]
Build NNPACK with ndk-build build system. Note: an NDK project must be built with Clang toolchain to use NNPACK as a static libray.

**Comment by Igamegum** [comment]
What Should I do in Fourth Step?

**Comment by Igamegum** [comment]
When I Run ndk-build ,I got the error message

`[x86] Compile++      : convolution-inference-alexnet-test <= alexnet.cc
[x86] Compile        : nnpack <= init.c
[x86] Compile        : nnpack <= convolution-output.c
[x86] Compile        : nnpack <= convolution-input-gradient.c
[x86] Compile        : nnpack <= convolution-kernel.c
[x86] Compile        : nnpack <= convolution-inference.c
[x86] Compile        : nnpack <= fully-connected-output.c
[x86] Compile        : nnpack <= fully-connected-inference.c
[x86] Compile        : nnpack <= pooling-output.c
[x86] Compile        : nnpack <= softmax-output.c
[x86] Compile        : nnpack <= relu-output.c
[x86] Compile        : nnpack <= relu-input-gradient.c
[x86] StaticLibrary  : libnnpack.a
[x86] Compile        : nnpack_reference <= convolution-output.c
[x86] Compile        : nnpack_reference <= convolution-input-gradient.c
[x86] Compile        : nnpack_reference <= convolution-kernel.c
[x86] Compile        : nnpack_reference <= fully-connected-output.c
[x86] Compile        : nnpack_reference <= max-pooling-output.c
[x86] Compile        : nnpack_reference <= softmax-output.c
[x86] Compile        : nnpack_reference <= relu-output.c
[x86] Compile        : nnpack_reference <= relu-input-gradient.c
[x86] StaticLibrary  : libnnpack_reference.a
[x86] Compile++      : gtest <= gtest-all.cc
[x86] StaticLibrary  : libgtest.a
[x86] Compile        : nnpack_ukernels <= 2d-fourier-8x8.c
[x86] Compile        : nnpack_ukernels <= 2d-fourier-16x16.c
[x86] Compile        : nnpack_ukernels <= 2d-winograd-8x8-3x3.c
[x86] Compile        : nnpack_ukernels <= relu.c
[x86] Compile        : nnpack_ukernels <= softmax.c
[x86] Compile        : nnpack_ukernels <= fft-block-mac.c
[x86] Compile        : nnpack_ukernels <= shdotxf.c
[x86] Compile        : nnpack_ukernels <= s4gemm.c
[x86] Compile        : nnpack_ukernels <= c4gemm.c
[x86] Compile        : nnpack_ukernels <= s4c2gemm.c
[x86] Compile        : nnpack_ukernels <= c4gemm-conjb.c
[x86] Compile        : nnpack_ukernels <= s4c2gemm-conjb.c
[x86] Compile        : nnpack_ukernels <= c4gemm-conjb-transc.c
[x86] Compile        : nnpack_ukernels <= s4c2gemm-conjb-transc.c
[x86] Compile        : nnpack_ukernels <= sgemm.c
[x86] Compile        : nnpack_ukernels <= sdotxf.c
[x86] StaticLibrary  : libnnpack_ukernels.a
[x86] Compile        : pthreadpool <= threadpool-pthreads.c
jni/../jni/../deps/pthreadpool/src/threadpool-pthreads.c:67:19: error: use of
      undeclared identifier 'SYS_futex'
                        return syscall(SYS_futex, address, FUTEX_WAIT_PR...
                                       ^
jni/../jni/../deps/pthreadpool/src/threadpool-pthreads.c:67:39: error: use of
      undeclared identifier 'FUTEX_WAIT_PRIVATE'
                        return syscall(SYS_futex, address, FUTEX_WAIT_PR...
                                                           ^
jni/../jni/../deps/pthreadpool/src/threadpool-pthreads.c:72:19: error: use of
      undeclared identifier 'SYS_futex'
                        return syscall(SYS_futex, address, FUTEX_WAKE_PR...
                                       ^
jni/../jni/../deps/pthreadpool/src/threadpool-pthreads.c:72:39: error: use of
      undeclared identifier 'FUTEX_WAKE_PRIVATE'
                        return syscall(SYS_futex, address, FUTEX_WAKE_PR...
                                                           ^
4 errors generated.
make: *** [obj/local/x86/objs/pthreadpool/jni/__/deps/pthreadpool/src/threadpool-pthreads.o] Error 1
`

**Comment by Maratyszcza** [comment]
The error is now fixed in Maratyszcza/pthreadpool@6615a1cd7fded042282387774c90538dc558affa

**Comment by Igamegum** [comment]
It Work,But how to used the compiled binary files？

**Comment by Maratyszcza** [comment]
@Igamegum NNPACK is not a deep learning framework. NNPACK does computations in neural network layers, but it doesn't load or store neural network layers, or manages computational graph. If you want to replay a pre-trained neural network with NNPACK acceleration, you need one of the deep learning frameworks that integrate NNPACK. I would recommend Caffe2, MXNet, or tiny-dnn.

**Comment by Igamegum** [comment]
Thanks！ @Maratyszcza
